pkg/cri/resource-manager/policy/flags.go

```diff
@@ -163,11 +165,28 @@ func (cs *ConstraintSet) parseCPUQuantity(value string) error {
 }
 
 func (cs *ConstraintSet) parseCPUFromCgroup(dir string) error {
-	cpusetDir := cgroups.Cpuset.Path()
-	if !strings.HasPrefix(dir, cpusetDir+"/") {
-		dir = filepath.Join(cpusetDir, dir)
+	ifExists := func(path string) string {
+		if _, err := os.Stat(path); errors.Is(err, os.ErrNotExist) {
+			return ""
+		}
+		return path
+	}
+	path := ""
+	if dir[0] == '/' {
+		// dir is a direct, absolute path to an existing cgroup
+		path = ifExists(filepath.Join(dir, "cpuset.cpus"))
+	}
+	if path == "" {
+		// dir is a relative path starting from the cgroup mount point
+		path = ifExists(filepath.Join(cgroups.GetMountDir(), dir, "cpuset.cpus"))
+	}
+	if path == "" {
+		// dir is a relative path starting from the cpuset controller (cgroup v1)
+		path = ifExists(filepath.Join(cgroups.Cpuset.Path(), dir, "cpuset.cpus"))
+	}
+	if path == "" {
+		return policyError("failed to find cgroup with cpuset.cpus for CPU cpuset constraint %q", dir)
 	}
```

Comment:
Looks basically fine to me.
nitpick: Why not an expressionless switch with equivalent cases than an effective if - elif - elif - elif - else -construct.

Reply:
Switched to `switch`.

Reply:
Sorry for my ignorance but is the switch working the same way now? I mean the original code tries to figure out the path multiple ways, but the switch is now doing things only once.